### PR TITLE
Add cmake build script for math library

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- fix: git status/git status --ignore normalization check problem 
- cmake:add kissfft CMake build
- cmake:add gemmlowp CMake build
- cmake:add ruy CMake build

## Impact

new cmake file

## Testing

ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)

__Fill In The Commit Message:__ This PR contains a Commit with an Empty Commit Message. Please fill in the Commit Message with the PR Summary.



**No, this PR does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:** 
    * Lacks a clear explanation of the problem being fixed. What is the "git status/git status --ignore normalization check problem"?
    * Needs more details on how the CMake builds for kissfft, gemmlowp, and ruy are implemented and their purpose within NuttX.
* **Impact:**
    *  While it mentions a new CMake file, it doesn't elaborate on the impact. 
    *  All other impact categories are left blank, indicating a lack of consideration for potential consequences.
* **Testing:**
    * Simply stating "ci" is insufficient. 
    * Provide specific details about the build hosts, target architectures/boards, and the actual testing logs before and after the change.

**Recommendations:**

1. **Expand the Summary:** Clearly articulate the issue being addressed and the implementation details of the CMake additions.
2. **Complete the Impact Section:** Address all impact categories, even if the answer is "NO" with a brief justification.
3. **Provide Detailed Testing Information:** List the build hosts, target environments, and include the actual testing logs before and after applying your changes.

File: math/gemmlowp/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  # ############################################################################
+  # Config and Fetch gemmlowp lib
+  # ############################################################################
+
+  set(GEMMLOWP_DIR ${CMAKE_CURRENT_LIST_DIR}/gemmlowp)
+
+  if(NOT EXISTS ${GEMMLOWP_DIR})
+    set(GEMMLOWP_URL
+        https://github.com/google/gemmlowp/archive/719139ce755a0f31cbf1c37f7f98adcc7fc9f425.zip
```
Comment (pkarashchenko) [inline math/gemmlowp/CMakeLists.txt:31]: Do we need to expose hash commit as a config option?

File: math/kissfft/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  # ############################################################################
+  # Config and Fetch kissfft lib
+  # ############################################################################
+
+  set(KISSFFT_DIR ${CMAKE_CURRENT_LIST_DIR}/kissfft)
+
+  if(NOT EXISTS ${KISSFFT_DIR})
+    set(KISSFFT_URL
+        https://github.com/mborgerding/kissfft/archive/refs/tags/v130.zip)
```
Comment (pkarashchenko) [inline math/kissfft/CMakeLists.txt:31]: Should version be exposed as configuration?

File: math/ruy/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+  # ############################################################################
+  # Config and Fetch ruy lib
+  # ############################################################################
+
+  set(RUY_DIR ${CMAKE_CURRENT_LIST_DIR}/ruy)
+
+  if(NOT EXISTS ${RUY_DIR})
+    set(RUY_URL
+        https://github.com/google/ruy/archive/d37128311b445e758136b8602d1bbd2a755e115d.zip
```
Comment (pkarashchenko) [inline math/ruy/CMakeLists.txt:31]: same

File: math/gemmlowp/CMakeLists.txt (hunk 1 above)
Comment (xiaoxiang781216) [reply math/gemmlowp/CMakeLists.txt:31]: we just test with this commit id, so its better to use the fix id until other people try the different version.

File: math/ruy/CMakeLists.txt (hunk 3 above)
Comment (xiaoxiang781216) [reply math/ruy/CMakeLists.txt:31]: ditto

File: math/kissfft/CMakeLists.txt (hunk 2 above)
Comment (xiaoxiang781216) [reply math/kissfft/CMakeLists.txt:31]: ditto

**Pull request merged**